timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockEventLogImpl.java

```diff
@@ -85,6 +86,7 @@ private Optional<LockWatchStateUpdate> tryGetNextEvents(Optional<LockWatchVersio
                 .map(events -> LockWatchStateUpdate.success(logId, slidingWindow.lastVersion(), events));
     }
 
+    @Unsafe
```

Comment:
Why is the annotation on `LockWatchStateUpdate` insufficient?

Reply:
I think there **isn't** one on LWSU currently: while the subtypes are _currently_ Unsafe, there's nothing stopping us from adding a Safe subtype, and so I think the decision was made that we do not propagate Unsafe/Safe-ness to a union type even if all of its member types are Unsafe.

Reply:
Yeah, that makes sense.